Rename Int63 ~> Uint63 (#13895)
coq/coq: doc/changelog/10-standard-library/13895-rename_int63.rst

- **Deprecated:**

  The naming convention for unsigned primitive integers is now ``uint63``
  instead of ``int63``.

>   Thus, the ``Int63.v`` file was deprecated and replaced by ``Uint63.v``,

Reviewer: ```suggestion
  The ``Int63`` module is replaced by ``Uint63``,
```

The title of the item already says "deprecated".  Users may think more in terms of modules rather than files--you think?

Author: Sure.

  ``int63_scope`` is now ``uint63_scope``, ``%int63`` is now ``%uint63``,
  ``Cyclic63.Int63Cyclic`` is now ``Cyclic63.Uint63Cyclic``, several terms in
  ``Ring63.v`` with the prefix ``Int63`` now have the prefix ``Uint63``,
  ``ZifyInt63.v`` is now ``ZifyUint63.v``, ``PrimFloat.of_int63`` is now
  ``PrimFloat.of_uint63``, and ``FloatAxioms.of_int63_spec`` is now
  ``FloatAxioms.of_uint63_spec``.

  Finally, the lemmas ``Z_lt_div2``, ``Zmod_le_first`` and ``negbE``
  (previously in ``Int63.v``) were deprecated in favor of
  ``Z.div_lt_upper_bound``, ``Z.mod_bound_pos_le`` and ``Bool.negb_sym``,

>   respectively.

Reviewer: If there are deprecation warnings for all of these, it may not be necessary to list them in detail.  Users will learn about the specific changes that affect them when they recompile their scripts.  Not sure giving the full details here is that helpful.

Author: I wondered about that. I'll delete this part and point to the first comment of this PR for the full list.

- **Deprecated:**

  Unsigned primitive integers are now named ``uint63`` instead of ``int63``.

>   The ``Int63`` module is replaced by ``Uint63``. For the full list of changes
>   see the first comment of the PR.

Reviewer: ```suggestion
  The ``Int63`` module is replaced by ``Uint63``. The full list of changes
  is described in the PR.
```

A minor distinction: IMHO the text block you submit when you create the PR is the "description".  Anything added separately after that is a "comment", part of the "conversation".
